multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -140,6 +173,18 @@ func (r *MemberClusterAnnounceReconciler) Reconcile(ctx context.Context, req ctr
 		// Nothing to handle in this case and MemberClusterAnnounce will also be deleted soon.
 		// TODO: Add ClusterClaim webhook to make sure it cannot be deleted while ClusterSet is present.
 	}
+
+	_, installed, _ := r.installedMemberClusterAnnounce.GetByKey(req.NamespacedName.String())
+	if !installed {
+		klog.InfoS("Adding member cluster to ClusterSet", "cluster", memberAnnounce.ClusterID)
+		if err := r.addMemberToClusterSet(memberAnnounce); err != nil {
+			klog.ErrorS(err, "Failed to add member cluster to ClusterSet", "cluster", memberAnnounce.ClusterID)
+			return ctrl.Result{}, err
+		}
+
+		r.installedMemberClusterAnnounce.Add(memberAnnounce)
+	}
+
 	// Member not found. If this happens, the MemberClusterAnnounce should soon be deleted.
```

Comment:
Please check and update the comments.

Reply:
Reminder. Probably we should move it to line 145, or better line 114 ("if !ok").
@luolanzone : please check this one too.